src/index.js

```diff
@@ -6,46 +6,52 @@ const tags = tagsW3.map(tag => tag.replace(/[<|>]/g, ""));
 
 // Wait for the DOM to be fully loaded
 document.addEventListener("DOMContentLoaded", function () {
-    uploadTagsFromLocalStorage(); // list all tags in localStorage
-    updateProgressBar();
-    // Find the form element and add a submit event listener
-    const form = document.querySelector("form");
-    form.addEventListener("submit", function (event) {
-        // on submit
-        event.preventDefault(); // Prevent the default form submission
-        submitVal(); // call our submitVal function
-    });
 
-    function submitVal() {
-        let val = document.getElementById("input1").value;
-        // remove all white spaces and non-alphabetical characters except . ! - 
-        let regex = /[^a-zA-Z!-.]/g;
-        // change input to lower case and apply regex
-        let sanitizedVal = val.toLowerCase().replace(regex, "");
+  uploadTagsFromLocalStorage(); // list all tags in localStorage
+  updateProgressBar();
+  // Find the form element and add a submit event listener
+  const form = document.querySelector("form");
+  form.addEventListener("submit", function (event) {
+    // on submit
+    event.preventDefault(); // Prevent the default form submission
+    submitVal(); // call our submitVal function
+  });
 
-        let li = "<li>" + sanitizedVal + "</li>";
-        // check if the input value is in the tags array
-        let found = tags.some(tag => tag.toLowerCase() === sanitizedVal);
-        if (found) {
-            document.getElementById("list").insertAdjacentHTML("beforeend", li);
-            saveTagToLocalStorage(sanitizedVal); // Save tag to localStorage
-            updateProgressBar();
-        }
-        document.getElementById("input1").value = "";
+  function submitVal() {
+    let val = document.getElementById("input1").value;
+    // remove all white spaces and non-alphabetical characters except . ! - 
+    let regex = /[^a-zA-Z!-.]/g;
+    // change input to lower case and apply regex
+    let sanitizedVal = val.toLowerCase().replace(regex, "");
+    let li = "<li>" + sanitizedVal + "</li>";
+    // check if the input value is in the tags array
+    let found = tags.some(tag => tag.toLowerCase() === sanitizedVal);
+    // get the list of inputs already made
+    const getList = localStorage.getItem("myTags");
+    let checkDuplicate = false;
```

Comment:
This would be my first choice for the name of the variable as well. But I've been told that the convention for a Boolean variable is to start with "is". Example, `isDuplicate`. I personally prefer the one we have now, but I just wanted to make you aware.

Reply:
I would use it with `isDuplicate` in React, I don't know why I didn't thought about it in JavaScript as well 🙈 .

Actually, I prefer using the `is` prefix. It makes it clear immediately that the output of the value is a `boolean`. I tried so many times using one of the already created variables, failed miserable, until I realised when I checked with `console.log()` that the values are different from what I expected.
I guess, when you are not that familiar with certain functions, it is not clear immediately (like `found` for example - I did not know what the expected output of `some()` is and assumed because of the naming that `found` contains all tags that were found in the list).

So, if you agree, I would add this to the JS refactor ticket!

Reply:
TIL: Oh, and I remebered the typeof operator after this discusssion and made use of it. Such an easy way to check the type of a value.